modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v1_x/FeatureTypeInfoImpl.java

```diff
@@ -77,7 +77,7 @@ public String getDescription() {
 
     @Override
     public String getName() {
-        return eType.getName().getLocalPart();
+        return eType.getName().getPrefix() + "_" + eType.getName().getLocalPart();
```

Comment:
Well first, the system as it exists today was not created by me and wfs-ng has always worked like this. But let me explain this: the prefix in the name is _not_ the prefix of the namespace of the resource in the geotools/geoserver consuming the wfs data. It is the prefix of the namespace on the wfs server. 

So for example, you have a WFS server with two layers:
blah:mylayer
bleh:myotherlayer

in geotools/geoserver you now create a wfs-ng datastore that feeds from this namespace, and the datastore is inside namespace 'ns'. You will now be able to make two layers:

ns:blah_mylayer
ns:bleh_myotherlayer

If you say that the resourceinfo.getName must correspond with DataStore.getFeatureSource( String) then that means my change is correct within the current logic.

It has always worked like this afaik.